### ISSUE TITLE
[Spree Upgrade] Fix bulk_line_item_controller spec (by defining a shipment in test line_items)

### ISSUE THREAD

**Issue by luisramos0** [openfoodfoundation/openfoodnetwork#2573]
Part of migrating shipping method to shipments, see larger scope in #2009.
Related to the move of inventory_unit.increase to stock_locations and shipments in #2559 
To fix the spec **bulk_line_items_controller_spec**, we need to provide the target_shipment in the test line_items.

The failed test message is:
 ```
1) Admin::BulkLineItemsController#index as a normal user should deny me access to the index action
     Failure/Error: let!(:line_item1) { FactoryBot.create(:line_item, order: order1) }
     NoMethodError:
       undefined method `stock_location' for nil:NilClass
     # /Users/luisramos/dev/ofn/spree/core/app/models/spree/order_inventory.rb:68:in `add_to_shipment'
     # /Users/luisramos/dev/ofn/spree/core/app/models/spree/order_inventory.rb:25:in `verify'
     # /Users/luisramos/dev/ofn/spree/core/app/models/spree/line_item.rb:90:in `update_inventory'
```

This is due to a new hook on spree line_item model: **before_save :update_inventory** that required a shipment to be available either at the line_item level (target_shipment, the one used in the solution, see PR for details) or at inventory level. See method OrderInventory.determine_target_shipment (this method is executed on line_item save, if there is no line_item.target shipment).

I believe this will have to be done in other specs... maybe we create other issues for that.

**Comment by luisramos0** [comment]
Also changed here is OFN's shipment decorator, it's changed to work with Shipments without adjustments. I am not sure this is correct BUT the reason I do this is that the spree code works with Shipments without Adjustments, it's just OFN's decorator that is failing.

Test failing message:
 ```
 4) Admin::BulkLineItemsController#index as an administrator when no ransack params are passed in formats final_weight_volume as a float
     Failure/Error: adjustment.set_included_tax! 0

     NoMethodError:
       undefined method `set_included_tax!' for nil:NilClass
     # ./app/models/spree/shipment_decorator.rb:9:in `ensure_correct_adjustment_with_included_tax'
```

**Comment by daniellemoorhead** [comment]
Closed this as #2574 is merged. @luisramos0 @mkllnk feel free to reopen if there's more to do...
